tools/c7n_azure/tests/test_template_utils.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2015-2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from azure_common import BaseTest
+from c7n_azure.session import Session
+from c7n_azure.template_utils import TemplateUtilities
+
+
+class TemplateUtilsTest(BaseTest):
+    def setUp(self):
+        super(TemplateUtilsTest, self).setUp()
+        self.template_util = TemplateUtilities()
+
+    def test_create_resource_group(self):
+        group_name = 'cloud-custodian-test'
+        self.template_util.create_resource_group(group_name, {'location': 'West US 2'})
+        s = Session()
+        #: :type: azure.mgmt.resource.ResourceManagementClient
+        client = s.client('azure.mgmt.resource.ResourceManagementClient')
+        resource_group = client.resource_groups.get(group_name)
+        self.assertIsNotNone(resource_group)
+
+    def test_deploy_template_with_parameters(self):
+        template_file = 'dedicated_functionapp.json'
+        group_name = 'cloud-custodian-test'
+        parameters = self.template_util.get_default_parameters(
+            'dedicated_functionapp.test.parameters.json')
+        self.template_util.deploy_resource_template(group_name, template_file, parameters)
```

Comment:
Does this assume the previous test ran to make RG? Not sure you get any guarantee of that.

Reply:
Good point, will fix this test.